teps/0141-platform-context-variables.md

```diff
@@ -221,190 +232,319 @@ The downside of this approach is that Tasks can't easily reference the values of
 
 ## Proposal
 
-<!--
-This is where we get down to the specifics of what the proposal actually is.
-This should have enough detail that reviewers can understand exactly what
-you're proposing, but should not include things like API designs or
-implementation. The "Design Details" section below is for the real
-nitty-gritty.
--->
+There are several components to this proposal:
 
-### Notes and Caveats
+- New TaskRun and PipelineRun syntax allowing TaskRun and PipelineRun authors
+to reference platform-provided context variables
+- The mechanism platform builders use to specify the values of each context variable per Run
 
-<!--
-(optional)
+## Design Details
 
-Go in to as much detail as necessary here.
-- What are the caveats to the proposal?
-- What are some important details that didn't come across above?
-- What are the core concepts and how do they relate?
--->
+### Stability level
 
+This feature will start as an alpha feature and will be enabled by a new feature flag "enable-platform-context-variables".
 
-## Design Details
+### TaskRun and PipelineRun syntax
 
-<!--
-This section should contain enough information that the specifics of your
-change are understandable. This may include API specs (though not always
-required) or even code snippets. If there's any ambiguity about HOW your
-proposal will be implemented, this is the place to discuss them.
-
-If it's helpful to include workflow diagrams or any other related images,
-add them under "/teps/images/". It's upto the TEP author to choose the name
-of the file, but general guidance is to include at least TEP number in the
-file name, for example, "/teps/images/NNNN-workflow.jpg".
--->
+This TEP proposes adding support for the context variable substitutions in TaskRuns and PipelineRuns of the form `$(context.platform.foo)`, where `foo` is the name of a parameter supplied by the platform. Platforms may provide string, array, or object parameters, so TaskRuns and PipelineRuns may also use substitutions like `$(context.platform.my-array-param[*])` or `$(context.platform.my-object-param.key)`.
```

Comment:
I'm not sure why do we call it $(context.platform.foo)?  Based on the below syntax and usage, I feel something like context.param.foo/ platform.param.foo could be better?

Reply:
I think it is because our existing context [variables](https://github.com/tektoncd/pipeline/blob/main/docs/variables.md), e.g. $(context.taskRun.name). So it has the context.level.name format